ARTICLE.md

```diff
@@ -1,59 +1,144 @@
+
+## What's in this guide
+
+- [Traditional unit tests based on examples](#traditional-unit-tests-based-on-examples)
+    - [Limitations of example-based testing](#limitations-of-example-based-testing)
+- [Introduction to property-based testing](#introduction-to-property-based-testing)
+    - [An example using Hypothesis](#an-example-using-hypothesis)
+    - [What can be a property?](#what-can-be-a-property)
+    - [How does property-based testing differ from example-based?](#how-does-property-based-testing-differ-from-example-based)
+- [Example properties and how to test for them](#example-properties-and-how-to-test-for-them)
+    - [Unexpected exceptions should never be thrown](#unexpected-exceptions-should-never-be-thrown)
+    - [Values shouldn't change after encoding and then decoding](#values-shouldnt-change-after-encoding-and-then-decoding)
+    - [A naive method should still give the same result](#a-naive-method-should-still-give-the-same-result)
+- [Finding bugs with property-based testing](#finding-bugs-with-property-based-testing)
+- [Available libraries](#available-libraries)
+- [Conclusion](#conclusion)
+
+⚠️ Prerequisites:
+- A general understanding of how software testing works.
+- (Optional) [Python 3+](https://www.python.org/downloads/)* if you want to follow along in your own IDE. 
+
+_* This guide will use Python for code examples, but the concepts aren't limited to Python specifically. So even if you don't know Python, we'd encourage you to read along anyway._
+
+## Traditional unit tests based on examples
+
+Most often, software testing is done using **example-based testing**. This means you test that for a given argument, you get a known return value. This return value is known because, well, you provided that value as a sample. So when the you run the function or test system, it then asserts the actual result against that sample return value.
+
+Let's look at an example. Say we wanted to write a function called `sort_this_list` that takes a [list](https://docs.python.org/3/tutorial/introduction.html#lists) as an argument and returns the same list organized in ascending order. To do this, we'll use the built-in Python [`sorted`](https://docs.python.org/3/library/functions.html#sorted) function.
+
+It might look like the following:
+
+```python
+def sort_this_list(list):
+    sorted_list = sorted(list)
+    return sorted_list
+```
+
+Now that we have our `sort_this_list` function, let's test it. 
+
+To test this using example-based testing, we need to provide the test function with return values that we know will be `True`. For example, the list `[5, 3, 1, 4, 2]` should return `[1, 2, 3, 4, 5]` after it's sorted.
+
+```python
+# Function that organizes list items in ascending order:
+def sort_this_list(list):
+    sorted_list = sorted(list)
+    return sorted_list
+
```

Comment:
> should we leave out the redefinition of the function here?

I was actually debating that while writing. Personally, when it comes to testing tutorials, I like seeing the function and test side-by-side to compare but it's not necessary...

I still want to remind people of the function later in the PBT section, but I think here we can remove it in this case because the original function is only a couple of lines up.